benchmark-service/benchmark/benchmark.py

```diff
@@ -1,48 +1,126 @@
+    benchmarks['data'] = sorted(data, key=lambda i: i['name'])
+    return benchmarks
+
 
 @router.get('/healthcheck')
 async def healthcheck(request: web.Request) -> web.Response:  # pylint: disable=unused-argument
     return web.Response()
 
 
-@router.get('/{username}')
-async def greet_user(request: web.Request) -> web.Response:
+@router.get('/name/{name}')
+@web_authenticated_developers_only(redirect=False)
+async def show_name(request: web.Request, userdata) -> web.Response:  # pylint: disable=unused-argument
+    query = request.query.get('file')
+    benchmarks = get_benchmarks(query)
+    # benchmarks = get_benchmarks(default_filepath)
+    name_data = next((item for item in benchmarks['data'] if item['name'] == str(request.match_info['name'])),
+                     None)
+
+    def get_x_list():
+        x_list = []
+        i = 0
+        for trial in name_data['trials']:
+            temp = []
+            temp = [i] * len(trial)
+            x_list.extend(temp)
+            i += 1
+        return x_list
+
+    x_list_test = get_x_list()  # This is a test
+    try:
+        fig = px.scatter(x=x_list_test, y=name_data['times'])
+        plot = json.dumps(fig, cls=plotly.utils.PlotlyJSONEncoder)
+    except Exception:
+        message = 'could not find name'
+        log.info('name is of type NoneType: ' + message, exc_info=True)
+        raise web.HTTPBadRequest(text=message)
 
     context = {
-        'username': request.match_info.get('username', ''),
-        'current_date': 'July 10, 2020'
+        'name': request.match_info.get('name', ''),
+        'plot': plot
     }
-    response = aiohttp_jinja2.render_template('user.html', request,
-                                              context=context)
-    return response
+
+    return await render_template('benchmark', request, userdata, 'name.html', context)
 
 
 @router.get('/')
 @router.get('')
 @web_authenticated_developers_only(redirect=False)
 async def index(request: web.Request, userdata) -> Dict[str, Any]:  # pylint: disable=unused-argument
-    context = {
-        'current_date': 'July 10, 2020'
-    }
-    response = aiohttp_jinja2.render_template('index.html', request,
-                                              context=context)
-    return response
+    benchmarks_context = get_benchmarks(default_filepath)
+    context = {'file': None,
+               'benchmarks': benchmarks_context}
+    return await render_template('benchmark', request, userdata, 'index.html', context)
+
+
+@router.get('/lookup')
+@web_authenticated_developers_only(redirect=False)
+async def lookup(request, userdata):  # pylint: disable=unused-argument
+    file = request.query.get('file')
+    if file is None:
+        return web.HTTPBadRequest()  # you'll need to look what this is up. Or you can set it to be the default file.
+    benchmarks_context = get_benchmarks(file)
+    context = {'file': file,
+               'benchmarks': benchmarks_context}
+
```

Comment:
I think the functionality in `/` and `/lookup` are basically duplicated except for the file. I think you can remove the lookup route and just have the index function look like this:

```
async def index(request, userdata):
     file = request.query.get('file')
     if file is None:
          file = default_filepath
     ....
```